specs/core/1_shard-data-chains.md

```diff
@@ -209,3 +272,275 @@ The `shard_chain_commitment` is only valid if it equals `compute_commitment(head
 ### Shard block fork choice rule
 
 The fork choice rule for any shard is LMD GHOST using the shard chain attestations of the persistent committee and the beacon chain attestations of the crosslink committee currently assigned to that shard, but instead of being rooted in the genesis it is rooted in the block referenced in the most recent accepted crosslink (ie. `state.crosslinks[shard].shard_block_root`). Only blocks whose `beacon_chain_ref` is the block in the main beacon chain at the specified `slot` should be considered (if the beacon chain skips a slot, then the block at that slot is considered to be the block in the beacon chain at the highest slot lower than a slot).
+
+# Updates to the beacon chain
+
+## Data structures
+
+### `Validator`
+
+Add member values to the end of the `Validator` object:
+
+```python
+    'open_branch_challenges': [BranchChallengeRecord],
+    'next_subkey_to_reveal': 'uint64',
+    'reveal_max_periods_late': 'uint64',
+```
+
+And the initializers:
+
+```python
+    'open_branch_challenges': [],
+    'next_subkey_to_reveal': 0,
+    'reveal_max_periods_late': 0,
+```
+
+Rename `withdrawal_epoch` to `withdrawable_epoch`.
```

Comment:
Another possibility is we get rid of the WITHDRAWABLE flag, and make the `withdrawal_epoch -> withdrawable_epoch` change in phase 0 (or, add both a withdrawal and withdrawable epoch), and move the change that penalized validators are not part of the queue to phase 0 as well. This would reduce total complexity including the complexity of the phase 0 -> phase 1 transition.

Reply:
See #615